### PR TITLE
lightning: "error" strategy outputs duplicate error in terminal, log and table

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/41629

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: br/tests/lightning_config_max_error/run.sh [hunk 1]
```diff
@@ -102,3 +102,15 @@ run_sql 'DROP DATABASE IF EXISTS mytest'
 cp "${mydir}/tidb-limit-record.toml" "${TEST_DIR}/tidb-limit-record.toml"
 sed -i.bak "s/threshold = 5/threshold = 4/g" "${TEST_DIR}/tidb-limit-record.toml"
 run_lightning --backend tidb --config "${TEST_DIR}/tidb-limit-record.toml" 2>&1 | grep -q "The number of conflict errors exceeds the threshold"
+
+# Check when strategy is "error", the stderr, log and duplicate record table all contains the error message
+run_sql 'DROP DATABASE IF EXISTS lightning_task_info'
+run_sql 'DROP DATABASE IF EXISTS mytest'
+rm "${TEST_DIR}/lightning.log"
+run_lightning --backend tidb --config "${mydir}/tidb-error.toml" 2>&1 | grep -q "Error 1062 (23000): Duplicate entry '1' for key 'testtbl.PRIMARY'"
+grep -q "Error 1062 (23000): Duplicate entry '1' for key 'testtbl.PRIMARY'" "${TEST_DIR}/lightning.log"
```
Comment (D3Hunter) [inline br/tests/lightning_config_max_error/run.sh:111]:
```suggestion
check_contains "Error 1062 (23000): Duplicate entry '1' for key 'testtbl.PRIMARY'" "${TEST_DIR}/lightning.log"
```

**Review by D3Hunter — APPROVED** [review]

**Comment by lance6716** [comment]
/hold

need address comment

**Pull request merged**